### PR TITLE
Add namecheap provider

### PR DESCRIPTION
**Pull request by nrfox**
Adds libdns provider for [namecheap](https://www.namecheap.com/support/api/intro/).

### CODE REVIEW

File: provider.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	// APIEndpoint to use. If testing, you can use the "sandbox" endpoint
+	// instead of the production one.
+	APIEndpoint string `json:"api_endpoint,omitempty"`
+
+	// ClientIP is the IP address of the requesting client.
+	// If this is not set, a discovery service will be
+	// used to determine the public ip of the machine.
+	// You must first whitelist your IP in the namecheap console
+	// before using the API.
+	ClientIP string `json:"client_IP,omitempty`
```
Comment (mholt) [inline provider.go:57]: Usualy convention like this would dictate `client_ip` (lowercase).

**Review by mholt — APPROVED** [review]
LGTM, thank you very much for contributing! I'll make repos in libdns and caddy-dns for you to push to. Remember to update the module paths and such.

File: provider.go (hunk 1 above)
Comment (nrfox) [reply provider.go:57]: Whoops that was a mistake

**Comment by nrfox** [comment]
> LGTM, thank you very much for contributing! I'll make repos in libdns and caddy-dns for you to push to. Remember to update the module paths and such.

Thank you for Caddy! It's one of those rare pieces of software that always just works out of the box and when you need to, takes very little time to configure. I really enjoy using it!

**Comment by mholt** [comment]
Thanks for the kind words! It's great to have people contributing back to help others too.

**Pull request closed without merging**